src/Parser/Warlock/Demonology/Modules/Features/DEUptimePet.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import SpellLink from 'common/SpellLink';
+import SpellIcon from 'common/SpellIcon';
+import {formatPercentage} from 'common/format';
+import StatisticBox from 'Main/StatisticBox';
+
+import SPELLS from 'common/SPELLS';
+
+
+const DE_DURATION = 12;
+const MILLISECONDS = 1000;
+
+class DEUptimePet extends Analyzer{
+
+  total_empowered_time = 0;
+  last_de_timestamp = null;
+
+  get uptime(){
+    return (this.total_empowered_time/this.owner.fightDuration);
+  }
+
+  get suggestionThresholds(){
+    return {
+      actual: this.uptime,
+      isLessThan: {
+        minor: 0.95,
+        average: 0.90,
+        major: 0.80,
+      },
+      style: 'percentage',
+    };
+  }
+
+  on_toPlayer_applybuff(event){
+    const spellId = event.ability.guid;
+    if(event.prepull && spellId === SPELLS.DEMONIC_EMPOWERMENT.id){
+      //Player may have pre-buffed their pet.
+      this.last_de_timestamp = this.owner.fight.start_time;
+    }
+  }
+
+  on_byPlayer_cast(event){
+    // We're making the assumption that the player has their permament pet active.
+    // Which as a Demonology warlock, we really hope is a reasonable assumption.
```

Comment:
How about making a module that tracks pet uptime at some point in the future? This could improve this accuracy and lead to a good suggestion since all reasonable assumptions are occasionally missed by players with bigger issues.